extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -465,16 +504,42 @@ export class AmpLightboxViewer extends AMP.BaseElement {
     this.updateInViewport(dev().assertElement(this.container_), true);
     this.scheduleLayout(dev().assertElement(this.container_));
 
-    this.currentElementId_ = element.lightboxItemId;
-    // Hack to access private property. Better than not getting
-    // type checking to work.
-    /**@type {?}*/ (this.carousel_).implementation_.showSlideWhenReady(
-        this.currentElementId_);
-
     this.win.document.documentElement.addEventListener(
         'keydown', this.boundHandleKeyboardEvents_);
 
-    return Promise.resolve();
+    if (this.carousel_) {
+      return this.resources_.requireLayout(this.carousel_)
+          .then(() => {
+            this.currentElemId_ = element.lightboxItemId;
+            // Hack to access private property. Better than not getting
+            // type checking to work.
+            /**@type {?}*/ (this.carousel_).implementation_.showSlideWhenReady(
+                this.currentElemId_);
+            this.resizeImageViewerDimensions_();
+            this.updateDescriptionBox_();
+          });
+    } else {
+      // We should never be opening a lightbox whose carousel hasn't built.
+      dev().error(TAG, 'Trying to open an uninitialized lightbox viewer.');
+      return Promise.resolve();
+    }
+
+  }
+
+  /**
+   * Resizes the ImageViewer of the current slide so that the image
+   * is centered in the page. Used in open or on slide change. Also
+   * registers a onResize handler to resize the ImageViewer whenever
+   * the screen size changes.
+   * @private
+   */
+  resizeImageViewerDimensions_() {
+    const imgViewer = this.elementsMetadata_[this.currentElemId_].imageViewer;
+    imgViewer.measure();
```

Comment:
better performance in a `vsync.measure()` block.
ditto below.

Reply:
Added `vsync.measure` in the `ImageViewer.measure` function.